File: packages/python/peru_spatial_utils/distance/distance.py
import math
from typing import Sequence

EARTH_RADIUS_KM = 6371.0088


def distance(a: Sequence[float], b: Sequence[float]) -> float:
    lng1, lat1 = a[0], a[1]
    lng2, lat2 = b[0], b[1]
    phi1 = math.radians(lat1)
    phi2 = math.radians(lat2)
    d_phi = math.radians(lat2 - lat1)
    d_lambda = math.radians(lng2 - lng1)
    h = (
        math.sin(d_phi / 2) ** 2
        + math.cos(phi1) * math.cos(phi2) * math.sin(d_lambda / 2) ** 2
    )
    return 2 * EARTH_RADIUS_KM * math.asin(math.sqrt(h))

File: packages/go/distance.go
package perspatial

import "math"

func Distance(a, b [2]float64) float64 {
	const r = 6371.0088
	lng1, lat1 := a[0], a[1]
	lng2, lat2 := b[0], b[1]

	phi1 := lat1 * math.Pi / 180
	phi2 := lat2 * math.Pi / 180
	dPhi := (lat2 - lat1) * math.Pi / 180
	dLambda := (lng2 - lng1) * math.Pi / 180

	h := math.Sin(dPhi/2)*math.Sin(dPhi/2) +
		math.Cos(phi1)*math.Cos(phi2)*math.Sin(dLambda/2)*math.Sin(dLambda/2)
	return 2 * r * math.Asin(math.Sqrt(h))
}
